src/main/scala/shine/DPIA/Phrases/Phrase.scala

```diff
@@ -130,43 +130,10 @@ object Phrase {
   def substitute[T1 <: PhraseType, T2 <: PhraseType](ph: Phrase[T1],
                                                      `for`: Phrase[T1],
                                                      in: Phrase[T2]): Phrase[T2] = {
-    var substCounter = 0
     object Visitor extends VisitAndRebuild.Visitor {
-      def renaming[X <: PhraseType](p: Phrase[X]): Phrase[X] = {
-        case class Renaming(idMap: Map[String, String]) extends VisitAndRebuild.Visitor {
-          override def phrase[T <: PhraseType](p: Phrase[T]): Result[Phrase[T]] = p match {
-            case Identifier(name, t) => Stop(
-              Identifier(idMap.getOrElse(name, name),
-                VisitAndRebuild.visitPhraseTypeAndRebuild(t, this)).asInstanceOf[Phrase[T]])
-            case l @ Lambda(x, _) =>
-              val newMap = idMap + (x.name -> freshName(x.name.takeWhile(_.isLetter)))
-              Continue(l, Renaming(newMap))
-            case dl @ DepLambda(_, x, _) =>
-              val newMap = idMap + (x.name -> freshName(x.name.takeWhile(_.isLetter)))
-              Continue(dl, Renaming(newMap))
-            case _ => Continue(p, this)
-          }
-
-          override def nat[N <: Nat](n: N): N = n.visitAndRebuild({
-            case i: NatIdentifier =>
-              NatIdentifier(idMap.getOrElse(i.name, i.name))
-            case ae => ae
-          }).asInstanceOf[N]
-
-          override def data[T <: DataType](dt: T): T = (dt match {
-            case i: DataTypeIdentifier =>
-              DataTypeIdentifier(idMap.getOrElse(i.name, i.name))
-            case dt => dt
-          }).asInstanceOf[T]
-        }
-        VisitAndRebuild(p, Renaming(Map()))
-      }
```

Comment:
I think that if we don't try to enforce unique names anymore, we should use proper capture-avoiding substitution here?

Reply:
@Bastacyclop can you create an issue for that, please?

Reply:
https://github.com/rise-lang/shine/issues/191